lms/djangoapps/discussion_api/tests/test_api.py

```diff
@@ -1240,6 +1283,8 @@ def test_basic(self, mock_emit):
             "endorsed_comment_list_url": None,
             "non_endorsed_comment_list_url": None,
             "editable_fields": ["abuse_flagged", "following", "raw_body", "title", "topic_id", "type", "voted"],
+            'read': False,
+            'has_endorsed': False
```

Comment:
Your first question can be explained in the discussion API proposal. https://openedx.atlassian.net/wiki/display/MA/Discussion+API+Proposal+Q4+2015

1. `has_endorsed` is whether the thread has comments that are endorsed.
`endorsed` is if the comment itself is endorsed.

2. That's the way we get the thread back from the comments client service.